### PR TITLE
add package filter feature

### LINKED ISSUE #1335: Status filter for package list
Add a new flag --status to filter pkg would be very helpful.
```
$ fission pkg list --status failed
```

### PR DESCRIPTION
**Pull request by MengZn**
Added a feature to filter the fission package ,Fixes #1335 
The results are as shown below.

```bash
# enable filter status 
fission pkg list --status failed
NAME                  BUILD_STATUS ENV
demo-src-pkg-zip-avjy failed       pythonsrc
```

```bash
# disable filter status 
fission pkg list
NAME                  BUILD_STATUS ENV
demo-src-pkg-zip-avjy failed       pythonsrc
hello-js-nv6k         succeeded    node
```

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1341)

### CODE REVIEW

File: pkg/fission-cli/package.go [hunk 1]
```diff
@@ -36,10 +37,8 @@ import (
 
 	"github.com/dchest/uniuri"
 	"github.com/fission/fission/pkg/utils"
-	"github.com/hashicorp/go-multierror"
 	"github.com/mholt/archiver"
 	"github.com/pkg/errors"
-	"github.com/satori/go.uuid"
 	"github.com/urfave/cli"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (life1347) [inline pkg/fission-cli/package.go]: Looks like the imports were missed deleted, please rollback the change.

File: pkg/fission-cli/main.go [hunk 2]
```diff
@@ -260,6 +260,7 @@ func NewCliApp() *cli.App {
 	pkgDeployArchiveFlag := cli.StringSliceFlag{Name: "deployarchive, deploy", Usage: "Local path or URL for binary archive"}
 	pkgBuildCmdFlag := cli.StringFlag{Name: "buildcmd", Usage: "Build command for builder to run with"}
 	pkgOutputFlag := cli.StringFlag{Name: "output, o", Usage: "Output filename to save archive content"}
+	pkgStatusFlag := cli.StringFlag{Name: "status", Usage: `filter packages status succeeded or failed"`}
```
Comment (life1347) [inline pkg/fission-cli/main.go]: Currently, fission supports not only succeeded/failed states. 
https://github.com/fission/fission/blob/b18462b10b600fe2de47c523a5e256039757f9f5/pkg/apis/fission.io/v1/const.go#L44-L48
So let's change the description a bit
```
Package status to filter packages  
```

Comment (MengZn) [reply pkg/fission-cli/main.go]: Is it better to use this statement?
```
Package status to filter packages
```

File: pkg/fission-cli/package.go (hunk 1 above)
Comment (MengZn) [reply pkg/fission-cli/package.go]: thank @life1347  for your review , that is my fail and I will fix this

File: pkg/fission-cli/main.go (hunk 2 above)
Comment (MengZn) [reply pkg/fission-cli/main.go]: I think what i mean here is to filter the packages by the package status, how about we change it to Filter package by status

Comment (life1347) [reply pkg/fission-cli/main.go]: That's fair. Let's do it. nitpick: `packages` -> `Filter packages by status`

**Comment by life1347** [comment]
LGTM

**Pull request merged**